modules/aaudio/player.c

```diff
@@ -0,0 +1,211 @@
+/**
+ * @file aaudio/player.c  AAudio audio driver for Android
+ *
+ * Copyright (C) 2024 Juha Heinanen
+ * Copyright (C) 2024 Sebastian Reimers
+ */
+
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+
+#include "aaudio.h"
+
+
+struct auplay_st {
+	auplay_write_h *wh;
+	void *arg;
+	struct auplay_prm play_prm;
+	size_t sampsz;
+};
+
+
+AAudioStream *playerStream = NULL;
```

Comment:
This should be part of the `auplay_st`, since with multiple calls or if a sound is played you have multiple streams to handle, same for recording. These have to be independent.

Reply:
How can I then close playerStream if starting of recorderStream fails and vice versa?

Reply:
Done by commit https://github.com/baresip/baresip/commit/8bce6730165288be5c5df5967e521a8f4bf670cc .